pallets/vc-management/src/lib.rs

```diff
@@ -218,19 +239,45 @@ pub mod pallet {
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
+		/// add an account to the delegatees
 		#[pallet::call_index(0)]
+		#[pallet::weight(<T as Config>::WeightInfo::add_delegatee())]
+		pub fn add_delegatee(origin: OriginFor<T>, account: T::AccountId) -> DispatchResult {
+			let _ = T::DelegateeAdminOrigin::ensure_origin(origin)?;
+			// we don't care if `account` already exists
+			Delegatee::<T>::insert(account.clone(), ());
+			Self::deposit_event(Event::DelegateeAdded { account });
+			Ok(())
+		}
+
+		/// remove an account from the delegatees
+		#[pallet::call_index(1)]
+		#[pallet::weight(<T as Config>::WeightInfo::remove_delegatee())]
+		pub fn remove_delegatee(origin: OriginFor<T>, account: T::AccountId) -> DispatchResult {
+			let _ = T::DelegateeAdminOrigin::ensure_origin(origin)?;
+			ensure!(Delegatee::<T>::contains_key(&account), Error::<T>::DelegateeNotExist);
```

Comment:
I'm a bit confused, `add_delegatee` is idempotent but `remove_delegatee` is not; either is fine I suppose but I'm not sure I see a reason for the difference :thinking:

Reply:
Maybe I don't get your point here but I was thinking adding delegatee can be "enforced" (so we don't care if the account is delegatee or not) but you can't remove it from delegatee if it's not in the list.

Maybe we could have saved one DB write if the account to be added is already a delegatee when `add_delegatee`, but we'll need another error. So personally I think either way is fine

Reply:
What I meant is, if you try adding the same delegatee twice then the worker doesn't complain, but if you try removing the same delegatee twice then it throws an error. Either way of handling it can be OK; I was just wondering why they're different from each other here :)